projects/habitat_ovmm/evaluator.py

```diff
@@ -168,12 +137,207 @@ def stop():
+            json.dump(aggregated_metrics, f, indent=4)
+        with open(f"{self.results_dir}/episode_results.json", "w") as f:
+            json.dump(episode_metrics, f, indent=4)
+
+    def local_evaluate(
+        self, agent, num_episodes: Optional[int] = None
+    ) -> Dict[str, float]:
+        if num_episodes is None:
+            num_episodes = self._env.number_of_episodes
+        else:
+            assert num_episodes <= self._env.number_of_episodes, (
+                "num_episodes({}) is larger than number of episodes "
+                "in environment ({})".format(num_episodes, self._env.number_of_episodes)
+            )
+
+        assert num_episodes > 0, "num_episodes should be greater than 0"
+
+        episode_metrics: Dict = {}
+
+        count_episodes: int = 0
+
+        pbar = tqdm(total=num_episodes)
+        while count_episodes < num_episodes:
+            observations, done = self._env.reset(), False
+            current_episode = self._env.get_current_episode()
+            agent.reset_vectorized([current_episode])
+
+            current_episode_key = (
+                f"{current_episode.scene_id.split('/')[-1].split('.')[0]}_"
+                f"{current_episode.episode_id}"
+            )
+            current_episode_metrics = {}
+
+            while not done:
+                action, info, _ = agent.act(observations)
+                observations, done, hab_info = self._env.apply_action(action, info)
+
+                if info["skill_done"] != "":
+                    metrics = self._extract_scalars_from_info(hab_info)
+                    metrics_at_skill_end = {
+                        f"{info['skill_done']}." + k: v for k, v in metrics.items()
+                    }
+                    current_episode_metrics = {
+                        **metrics_at_skill_end,
+                        **current_episode_metrics,
+                    }
+                    current_episode_metrics["goal_name"] = info["goal_name"]
+
+            metrics = self._extract_scalars_from_info(hab_info)
+            metrics_at_episode_end = {"END." + k: v for k, v in metrics.items()}
+            current_episode_metrics = {
+                **metrics_at_episode_end,
+                **current_episode_metrics,
+            }
+            current_episode_metrics["goal_name"] = info["goal_name"]
+
+            episode_metrics[current_episode_key] = current_episode_metrics
+            if len(episode_metrics) % self.metrics_save_freq == 0:
+                aggregated_metrics = self._aggregare_metrics(episode_metrics)
+                self._write_results(episode_metrics, aggregated_metrics)
+
+            count_episodes += 1
+            pbar.update(1)
+
+        self._env.close()
+        aggregated_metrics = self._aggregare_metrics(episode_metrics)
+        self._write_results(episode_metrics, aggregated_metrics)
+
+        return aggregated_metrics
+
+    def remote_evaluate(self, agent, num_episodes: Optional[int] = None):
```

Comment:
This function is copied from `habitat.core.benchmark.py` and needs to be updated.